clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2737,6 +2754,16 @@ void SYCLIntegrationHeader::emit(raw_ostream &O) {
 
   O << "\n";
 
+  int cpp_version = getCppVersion();
+  if (cpp_version != -1) {
+    O << "#define STD_CPP_VERSION ";
+    O << cpp_version << "\n";
+    O << "#if __cplusplus != STD_CPP_VERSION\n";
```

Comment:
This is actually pretty risky when using other compilers.  MSVC for example used a 1997 date until just recently.

Reply:
Handling windows resolve and/or linux specifiers in a separate issue/ticket